classes/PaymentModule.php

```diff
@@ -590,19 +590,14 @@ public function validateOrder(
 
                         // Join PDF invoice
                         if ((int) Configuration::get('PS_INVOICE') && $order_status->invoice && $order->invoice_number) {
-                            $currentLanguage = $this->context->language;
-                            $this->context->language = $orderLanguage;
-                            $this->context->getTranslator()->setLocale($orderLanguage->locale);
                             $order_invoice_list = $order->getInvoicesCollection();
                             Hook::exec('actionPDFInvoiceRender', ['order_invoice_list' => $order_invoice_list]);
-                            $pdf = new PDF($order_invoice_list, PDF::TEMPLATE_INVOICE, $this->context->smarty);
-                            $file_attachement['content'] = $pdf->render(false);
-                            $file_attachement['name'] = Configuration::get('PS_INVOICE_PREFIX', (int) $order->id_lang, null, $order->id_shop) . sprintf('%06d', $order->invoice_number) . '.pdf';
-                            $file_attachement['mime'] = 'application/pdf';
-                            $this->context->language = $currentLanguage;
-                            $this->context->getTranslator()->setLocale($currentLanguage->locale);
+                            $pdf = new PDF($order_invoice_list, PDF::TEMPLATE_INVOICE, $this->context->smarty, 'P', $orderLanguage);
+                            $file_attachment['content'] = $pdf->render(false);
```

Comment:
it's usually a good practice to declare an array before using it

```suggestion
                            $file_attachment = [];
                            $file_attachment['content'] = $pdf->render(false);
```

(or did I miss a previous declaration?)

Reply:
@matthieu-rolland You didn't ;-)